code/iaas/storage-service/src/main/java/io/cattle/platform/storage/api/filter/ExternalTemplateInstanceFilter.java

```diff
@@ -79,51 +83,33 @@ protected static void validateImageUuid(String image) {
         }
 
         String server = dockerImage.getServer();
-        String qualifiedName = dockerImage.getQualifiedName();
-        String whitelist = WHITELIST_REGISTRIES.get();
+        List<String> whitelistRegistries = WHITELIST_REGISTRIES.get();
         String defaultRegistry = DEFAULT_REGISTRY.get();
         String userProvidedRegistry = dockerImage.getServer();
 
-        if (!qualifiedName.startsWith(server)) {
+        if (!image.contains(server)) {
```

Comment:
Nit Could you also use consistent names for registry? Like here, "server" is really a registry, but it takes time to figure it out :)